language/move-ir/types/src/location.rs

```diff
@@ -8,23 +8,25 @@ use std::{
     fmt,
     hash::{Hash, Hasher},
 };
+use symbol_pool::Symbol;
 
 //**************************************************************************************************
 // Loc
 //**************************************************************************************************
 
+
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize, Hash)]
 pub struct Loc {
-    file: &'static str,
+    file: Symbol,
     span: Span,
 }
 impl Loc {
-    pub fn new(file: &'static str, span: Span) -> Loc {
-        Loc { file, span }
+    pub fn new(file: &str, span: Span) -> Loc {
+        Loc { file: Symbol::from(file), span }
```

Comment:
@tnowacki As we discussed, I'll move this `Symbol::from` conversion out of `Loc` itself and into the higher-level modules, like the source language and IR compilers.